### ISSUE TITLE
GLPI 10.0.7 : queuednotification growing

### ISSUE THREAD

**Issue by Benoit221** [glpi-project/glpi#14661]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

10.0.7

### Bug description

Since we upgraded to GLPI 10, we had trouble with notification.

Our queuednotification task is set to CLI and cron works perfectly with user apache (alpine based).
Event if i run manually php front/cron.php in command line with apache user, i see that the last execution is "now" but in queued notification there is 0 try to send them.
Notification are successfully send if i manually sent them from the queued notification menu.

Best regards

### Relevant log output

```shell
glpi-cron  | 2023-05-04T16:09:00.243341206Z crond: USER apache pid 4155 cmd /usr/bin/php /var/www/glpi/front/cron.php
glpi-cron  | 2023-05-04T16:10:00.244529371Z crond: USER apache pid 4156 cmd /usr/bin/php /var/www/glpi/front/cron.php
glpi-cron  | 2023-05-04T16:11:00.245840238Z crond: USER apache pid 4157 cmd /usr/bin/php /var/www/glpi/front/cron.php
glpi-cron  | 2023-05-04T16:12:00.246903902Z crond: USER apache pid 4158 cmd /usr/bin/php /var/www/glpi/front/cron.php
glpi-cron  | 2023-05-04T16:13:00.247770163Z crond: USER apache pid 4159 cmd /usr/bin/php /var/www/glpi/front/cron.php
glpi-cron  | 2023-05-04T16:14:00.248728525Z crond: USER apache pid 4160 cmd /usr/bin/php /var/www/glpi/front/cron.php
glpi-cron  | 2023-05-04T16:15:00.249593987Z crond: USER apache pid 4161 cmd /usr/bin/php /var/www/glpi/front/cron.php
glpi-cron  | 2023-05-04T16:15:00.250371496Z crond: USER root pid 4162 cmd run-parts /etc/periodic/15min
glpi-cron  | 2023-05-04T16:16:00.251408760Z crond: USER apache pid 4163 cmd /usr/bin/php /var/www/glpi/front/cron.php
glpi-cron  | 2023-05-04T16:17:00.252714326Z crond: USER apache pid 4164 cmd /usr/bin/php /var/www/glpi/front/cron.php
glpi-cron  | 2023-05-04T16:18:00.253612388Z crond: USER apache pid 4165 cmd /usr/bin/php /var/www/glpi/front/cron.php

2023-05-04 16:15:00 [@f9441e9655ed]
Externe #3 : Démarrage taskscheduler
2023-05-04 16:16:00 [@f9441e9655ed]
Externe #1 : Démarrage mailgate
2023-05-04 16:16:00 [@f9441e9655ed]
Externe #2 : Démarrage queuednotification
2023-05-04 16:16:00 [@f9441e9655ed]
Externe #3 : Démarrage taskscheduler
2023-05-04 16:16:00 [@f9441e9655ed]
Externe #4 : Démarrage wakeupAgents
2023-05-04 16:17:00 [@f9441e9655ed]
Externe #1 : Démarrage mailgate
2023-05-04 16:17:00 [@f9441e9655ed]
Externe #2 : Démarrage queuednotification
2023-05-04 16:17:00 [@f9441e9655ed]
Externe #3 : Démarrage taskscheduler
2023-05-04 16:18:00 [@f9441e9655ed]
Externe #1 : Démarrage mailgate
2023-05-04 16:18:00 [@f9441e9655ed]
Externe #2 : Démarrage queuednotification
2023-05-04 16:18:00 [@f9441e9655ed]
Externe #3 : Démarrage taskscheduler
2023-05-04 16:18:00 [@f9441e9655ed]
Externe #4 : Démarrage wakeupAgents
```

### Your GLPI setup information

<details><summary>Informations sur le système, l'installation et la configuration</summary><pre>GLPI 10.0.7 ( => /var/www/glpi)
Installation mode: TARBALL
Current language:fr_FR

</pre></details><details><summary>Server</summary><pre> 
Operating system: Linux 700d975369d4 5.10.0-21-amd64 #1 SMP Debian 5.10.162-1 (2023-01-21) x86_64
PHP 8.1.18 apache2handler (Core, PDO, Phar, Reflection, SPL, SimpleXML, Zend OPcache, apache2handler, apcu, bz2, ctype, curl,
	date, dom, exif, fileinfo, filter, gd, hash, iconv, igbinary, imap, intl, json, ldap, libxml, mbstring, mysqli, mysqlnd,
	openssl, pcre, pdo_mysql, pspell, redis, session, sodium, standard, xml, zip, zlib)
Setup: max_execution_time="30" memory_limit="256M" post_max_size="40M" safe_mode="" session.save_handler="files"
	upload_max_filesize="20M" 
Software: Apache/2.4.57 (Unix) (Apache/2.4.57 (Unix) Server at removed_for_confidentiality Port 80
)
	Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36 Edg/112.0.1722.68
Server Software: mariadb.org binary distribution
	Server Version: 10.11.2-MariaDB-1:10.11.2+maria~ubu2204
	Server SQL Mode: STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION
	Parameters: glpi-user@db/glpi
	Host info: db via TCP/IP
	
PHP version (8.1.18) is supported.
Sessions configuration is OK.
Allocated memory is sufficient.
mysqli extension is installed.
Following extensions are installed: dom, fileinfo, json, simplexml.
curl extension is installed.
gd extension is installed.
intl extension is installed.
libxml extension is installed.
zlib extension is installed.
The constant SODIUM_CRYPTO_AEAD_XCHACHA20POLY1305_IETF_NPUBBYTES is present.
Database engine version (10.11.2) is supported.
No files from previous GLPI version detected.
The log file has been created successfully.
Write access to /var/lib/glpi/_cache has been validated.
Write access to /etc/glpi has been validated.
Write access to /var/lib/glpi/_cron has been validated.
Write access to /var/lib/glpi has been validated.
Write access to /var/lib/glpi/_dumps has been validated.
Write access to /var/lib/glpi/_graphs has been validated.
Write access to /var/lib/glpi/_lock has been validated.
Write access to /var/lib/glpi/_pictures has been validated.
Write access to /var/lib/glpi/_plugins has been validated.
Write access to /var/lib/glpi/_rss has been validated.
Write access to /var/lib/glpi/_sessions has been validated.
Write access to /var/lib/glpi/_tmp has been validated.
Write access to /var/lib/glpi/_uploads has been validated.

Web server root directory configuration seems safe.
Sessions configuration is secured.
OS and PHP are relying on 64 bits integers.
exif extension is installed.
ldap extension is installed.
openssl extension is installed.
Following extensions are installed: bz2, Phar, zip.
Zend OPcache extension is installed.
Following extensions are installed: ctype, iconv, mbstring, sodium.
Write access to /var/www/glpi/marketplace has been validated.
Timezones seems loaded in database.

</pre></details><details><summary>GLPI constants</summary><pre> 
GLPI_ROOT: "/var/www/glpi"
GLPI_CONFIG_DIR: "/etc/glpi"
GLPI_VAR_DIR: "/var/lib/glpi"
GLPI_LOG_DIR: "/var/log/glpi"
GLPI_MARKETPLACE_DIR: "/var/www/glpi/marketplace"
GLPI_USE_CSRF_CHECK: "1"
GLPI_CSRF_EXPIRES: "7200"
GLPI_CSRF_MAX_TOKENS: "100"
GLPI_USE_IDOR_CHECK: "1"
GLPI_IDOR_EXPIRES: "7200"
GLPI_ALLOW_IFRAME_IN_RICH_TEXT: false
GLPI_SERVERSIDE_URL_ALLOWLIST: ["/^(https?|feed):\\/\\/[^@:]+(\\/.*)?$/"]
GLPI_TELEMETRY_URI: "https://telemetry.glpi-project.org"
GLPI_INSTALL_MODE: "TARBALL"
GLPI_NETWORK_MAIL: "glpi@teclib.com"
GLPI_NETWORK_SERVICES: "https://services.glpi-network.com"
GLPI_MARKETPLACE_ALLOW_OVERRIDE: true
GLPI_MARKETPLACE_MANUAL_DOWNLOADS: true
GLPI_USER_AGENT_EXTRA_COMMENTS: ""
GLPI_DISABLE_ONLY_FULL_GROUP_BY_SQL_MODE: "1"
GLPI_AJAX_DASHBOARD: "1"
GLPI_CALDAV_IMPORT_STATE: 0
GLPI_DEMO_MODE: "0"
GLPI_CENTRAL_WARNINGS: "1"
GLPI_DOC_DIR: "/var/lib/glpi"
GLPI_CACHE_DIR: "/var/lib/glpi/_cache"
GLPI_CRON_DIR: "/var/lib/glpi/_cron"
GLPI_DUMP_DIR: "/var/lib/glpi/_dumps"
GLPI_GRAPH_DIR: "/var/lib/glpi/_graphs"
GLPI_LOCAL_I18N_DIR: "/var/lib/glpi/_locales"
GLPI_LOCK_DIR: "/var/lib/glpi/_lock"
GLPI_PICTURE_DIR: "/var/lib/glpi/_pictures"
GLPI_PLUGIN_DOC_DIR: "/var/lib/glpi/_plugins"
GLPI_RSS_DIR: "/var/lib/glpi/_rss"
GLPI_SESSION_DIR: "/var/lib/glpi/_sessions"
GLPI_TMP_DIR: "/var/lib/glpi/_tmp"
GLPI_UPLOAD_DIR: "/var/lib/glpi/_uploads"
GLPI_INVENTORY_DIR: "/var/lib/glpi/_inventories"
GLPI_NETWORK_REGISTRATION_API_URL: "https://services.glpi-network.com/api/registration/"
GLPI_MARKETPLACE_PLUGINS_API_URI: "https://services.glpi-network.com/api/glpi-plugins/"
GLPI_I18N_DIR: "/var/www/glpi/locales"
GLPI_VERSION: "10.0.7"
GLPI_SCHEMA_VERSION: "10.0.7@5d45269702917a32805e25b678f6779a98b145f6"
GLPI_MARKETPLACE_PRERELEASES: false
GLPI_MIN_PHP: "7.4.0"
GLPI_MAX_PHP: "8.3.0"
GLPI_YEAR: "2023"

</pre></details><details><summary>Libraries</summary><pre> 
htmlawed/htmlawed version 1.2.9 in (/var/www/glpi/vendor/htmlawed/htmlawed)
phpmailer/phpmailer version 6.8.0 in (/var/www/glpi/vendor/phpmailer/phpmailer/src)
simplepie/simplepie version 1.5.8 in (/var/www/glpi/vendor/simplepie/simplepie/library)
tecnickcom/tcpdf version 6.6.2 in (/var/www/glpi/vendor/tecnickcom/tcpdf)
michelf/php-markdown in (/var/www/glpi/vendor/michelf/php-markdown/Michelf)
true/punycode in (/var/www/glpi/vendor/true/punycode/src)
iamcal/lib_autolink in (/var/www/glpi/vendor/iamcal/lib_autolink)
sabre/dav in (/var/www/glpi/vendor/sabre/dav/lib/DAV)
sabre/http in (/var/www/glpi/vendor/sabre/http/lib)
sabre/uri in (/var/www/glpi/vendor/sabre/uri/lib)
sabre/vobject in (/var/www/glpi/vendor/sabre/vobject/lib)
laminas/laminas-i18n in (/var/www/glpi/vendor/laminas/laminas-i18n/src)
laminas/laminas-servicemanager in (/var/www/glpi/vendor/laminas/laminas-servicemanager/src)
monolog/monolog in (/var/www/glpi/vendor/monolog/monolog/src/Monolog)
sebastian/diff in (/var/www/glpi/vendor/sebastian/diff/src)
donatj/phpuseragentparser in (/var/www/glpi/vendor/donatj/phpuseragentparser/src/UserAgent)
elvanto/litemoji in (/var/www/glpi/vendor/elvanto/litemoji/src)
symfony/console in (/var/www/glpi/vendor/symfony/console)
scssphp/scssphp in (/var/www/glpi/vendor/scssphp/scssphp/src)
laminas/laminas-mail in (/var/www/glpi/vendor/laminas/laminas-mail/src/Protocol)
laminas/laminas-mime in (/var/www/glpi/vendor/laminas/laminas-mime/src)
rlanvin/php-rrule in (/var/www/glpi/vendor/rlanvin/php-rrule/src)
blueimp/jquery-file-upload in (/var/www/glpi/vendor/blueimp/jquery-file-upload/server/php)
ramsey/uuid in (/var/www/glpi/vendor/ramsey/uuid/src)
psr/log in (/var/www/glpi/vendor/psr/log/Psr/Log)
psr/simple-cache in (/var/www/glpi/vendor/psr/simple-cache/src)
psr/cache in (/var/www/glpi/vendor/psr/cache/src)
league/csv in (/var/www/glpi/vendor/league/csv/src)
mexitek/phpcolors in (/var/www/glpi/vendor/mexitek/phpcolors/src/Mexitek/PHPColors)
guzzlehttp/guzzle in (/var/www/glpi/vendor/guzzlehttp/guzzle/src)
guzzlehttp/psr7 in (/var/www/glpi/vendor/guzzlehttp/psr7/src)
glpi-project/inventory_format in (/var/www/glpi/vendor/glpi-project/inventory_format/lib/php)
wapmorgan/unified-archive in (/var/www/glpi/vendor/wapmorgan/unified-archive/src)
paragonie/sodium_compat in (/var/www/glpi/vendor/paragonie/sodium_compat/src)
symfony/cache in (/var/www/glpi/vendor/symfony/cache)
html2text/html2text in (/var/www/glpi/vendor/html2text/html2text/src)
symfony/css-selector in (/var/www/glpi/vendor/symfony/css-selector)
symfony/dom-crawler in (/var/www/glpi/vendor/symfony/dom-crawler)
twig/twig in (/var/www/glpi/vendor/twig/twig/src)
twig/string-extra in (/var/www/glpi/vendor/twig/string-extra)
symfony/polyfill-ctype not found
symfony/polyfill-iconv not found
symfony/polyfill-mbstring not found
symfony/polyfill-php80 not found
symfony/polyfill-php81 not found
symfony/polyfill-php82 in (/var/www/glpi/vendor/symfony/polyfill-php82)
league/oauth2-client in (/var/www/glpi/vendor/league/oauth2-client/src/Provider)
league/oauth2-google in (/var/www/glpi/vendor/league/oauth2-google/src/Provider)
thenetworg/oauth2-azure in (/var/www/glpi/vendor/thenetworg/oauth2-azure/src/Provider)
phpCas version 1.3.8 in (/usr/share/php81)

</pre></details><details><summary>LDAP directories</summary><pre> 
Server: 'ldap://removed_for_confidentiality', Port: '389', BaseDN: 'removed_for_confidentiality', Connection filter:
		'(&(objectclass=user)(objectcategory=person)(!(useraccountcontrol:1.2.840.113556.1.4.803:=2))(!(objectclass=computer)))',
		RootDN: 'removed_for_confidentiality', Use TLS: none

</pre></details><details><summary>SQL replicas</summary><pre> 
Not active

</pre></details><details><summary>Notifications</summary><pre> 
Way of sending emails: SMTP (removed_for_confidentiality)

</pre></details><details><summary>Plugins list</summary><pre> 
	fields               Name: Champs supplémentaires         Version: 1.20.5     State: Enabled                                 
		Install Method: Marketplace
	datainjection        Name: Data Injection                 Version: 2.13.0     State: Enabled                                 
		Install Method: Marketplace
	glpiinventory        Name: GLPI Inventory                 Version: 1.2.1      State: Enabled                                 
		Install Method: Marketplace

</pre></details>

**Comment by Benoit221** [comment]
Some screenshot are in localtime other in UTC.

![image](https://user-images.githubusercontent.com/56193156/236266351-858b8fd7-e7a9-4712-8c4e-c691ee80e4e9.png)

Sometime one notification is sent
![image](https://user-images.githubusercontent.com/56193156/236266073-6b8760bd-aceb-4160-bc8c-3967d605977b.png)

![image](https://user-images.githubusercontent.com/56193156/236266753-1c80623a-8a5e-4b45-acbe-1b1606685f24.png)

In most case they are cleaned 

![image](https://user-images.githubusercontent.com/56193156/236267033-c5cb5833-d26a-4d7d-9e4e-28df65f3a501.png)

**Comment by skynetbilgi** [comment]
I also have the same problem. Did you find a solution?

**Comment by skynetbilgi** [comment]
**I upgraded glpi to version 10.0.7. Then mailgate is not working. I click the play button but it doesn't deliver any notification. When I select and send in the notification queue, it sends notifications manually.**

**Comment by cedric-anne** [comment]
> **I upgraded glpi to version 10.0.7. Then mailgate is not working. I click the play button but it doesn't deliver any notification. When I select and send in the notification queue, it sends notifications manually.**

- `mailgate` is the task that collect emails fomr configured mail receivers in order to create tickets.
- `queuednotification` is the task that sends notifications.

Each one is independent from the other one.

**Comment by keguira** [comment]
Quick question : why change the default value from "empty" with NULL (impliying "undefined") to 0.0 which is a real existing coordinate and sometimes used as an "error" value or used by people as default coordinates to not share it ?

**Comment by auguzsto** [comment]
@Benoit221 

I went through something similar and made some changes to the settings separating CLI and GLPI mode responsibilities.
- Automatic actions that are under 3 minutes are performed by CLI.
- Automatic actions that are above 3 minutes are executed by GLPI.
- Leave activated only the automatic actions that are being used.
- I no longer use crontab for automatic actions that use the CLI, I developed a basic script to run the command every minute.
- Do not run auto action command in CLI in general. That is, specify which automatic action it should run. Example: /usr/bin/php /var/www/html/glpi/front/cron.php queuednotification &>/dev/null;

These good practices improve my automatic actions by 100% and this type of problem has stopped.
